### ISSUE TITLE
Visual appearance of laissez vibrer

### ISSUE THREAD

**Issue by rettinghaus** [music-encoding/music-encoding#711]
We have a `lv` element, but this is AFAICS just for encoding an open tie. There should be an `@form` attribute to specific the visual appearance. 
Proposed values: 

- `"none"` - not visual, for semantic/audial encoding (e.g. for MIDI conversion)
- `"tie"` - _regular_ appearance as tie
- `"text"` - printed as _l.v._

**Comment by bwbohl** [comment]
Good suggestion, a value indicating both should also be available

**Comment by craigsapp** [comment]
Issue https://github.com/rism-ch/verovio/issues/92 has some examples of  laissez vibrer:


<img width="274" alt="screen shot 2019-02-08 at 8 25 32 am" src="https://user-images.githubusercontent.com/3487289/52480811-be497380-2b7a-11e9-9b06-a2f7ef551ebc.png">


from Debussy preludes book II: http://conquest.imslp.info/files/imglnks/usimg/d/dd/IMSLP00510-Debussy_-_Preludes,_Book_2.pdf:

<img width="495" alt="screen shot 2019-02-08 at 3 28 30 pm" src="https://user-images.githubusercontent.com/3487289/52504257-de4a5880-2bb5-11e9-9c30-ca02f3341e6c.png">


<img width="524" alt="screen shot 2019-02-08 at 3 34 16 pm" src="https://user-images.githubusercontent.com/3487289/52504512-b14a7580-2bb6-11e9-98a2-43e5a0a8a24a.png">

<img width="253" alt="screen shot 2019-02-08 at 3 39 09 pm" src="https://user-images.githubusercontent.com/3487289/52504766-6715c400-2bb7-11e9-87df-cae1a42f4d12.png">

<img width="499" alt="screen shot 2019-02-08 at 3 41 46 pm" src="https://user-images.githubusercontent.com/3487289/52504835-b0661380-2bb7-11e9-9454-186dfd5d775e.png">

<img width="527" alt="screen shot 2019-02-08 at 3 47 44 pm" src="https://user-images.githubusercontent.com/3487289/52505154-a1339580-2bb8-11e9-8bb3-623b948d24d6.png">

**Comment by lpugin** [comment]
One point regarding the textual versions. I wonder if it would make sense to allow `<lv>` to contain textual content. Then we could have things like `<lv>(laissez vibrer)</lv>` for cases were a specific text is desired. Of course, this does not mean that we could not support `<lv form="text"/>` without further indication, in which case `l.v.` could be displayed. However, I am not too sure about this. I don't think we do this elsewhere in MEI, so I would prefer not to have `text` as a value for `@form`. This would also avoid the need for an indication for both since you would simply do `<lv form="tie">l.v.</lv>`

(Obviously, one can argue that the text version is a `<dir>`. In that case, I think the additional an attribute on `<dir>` for qualifying directives, which was already discussed many times, would be desirable.)

**Comment by bwbohl** [comment]
actually I just realised there is a remark concerning the l.v. text on https://music-encoding.org/guidelines/v4/elements/lv.html maybe allowing `dir` inside `lv` (alongside `curve`which is already possible) could be an option, too. then one could either use the attributes to specify the appearance or use the contained elements if e.g. editorial interaction is to be expressed.

**Comment by pe-ro** [comment]
`<dir>` is allowed within `<lv>` to capture the text.  Putting text directly inside `<lv>` means that `<lv>` would have to take on all the attributes of `<dir>` regarding placement.  There would also most likely be a great deal of conflict if this were done because one set of attributes would be needed for positioning of the graphical sign (the slur-like curve) and for positioning of the text.  Better to put the text in `<dir>`.  The `<dir>` element doesn't have to be inside `<lv>`, but if it is, then there's no need for some kind of mechanism to link the `<lv>` and its associated `<dir>`.

**Comment by pe-ro** [comment]
There's no need for a `@form` attribute.  An `<lv>` will either have associated text (a `<dir>`) or it won't.  It may seem counterintutitive, but for a text-only laissez-vibrer, there will only be a `<dir>`, not an `<lv>`.  When only the text "l.v." (or similar) appears, there is, in fact, only a directive.  One may use `@type` to indicate the function of the `<dir>` element.

**Comment by lpugin** [comment]
> There's no need for a `@form` attribute. An `<lv>` will either have associated text (a `<dir>`) or it won't. It may seem counterintutitive, but for a text-only laissez-vibrer, there will only be a `<dir>`, not an `<lv>`. When only the text "l.v." (or similar) appears, there is, in fact, only a directive. One may use `@type` to indicate the function of the `<dir>` element.

This make sense. However, I think the idea to have something more structured than `@type` to indicate a function of a `<dir>` has been expressed many times in the community. This example is a good illustration of such a need in my opinion.

**Comment by kepper** [comment]
Agreed. And I think it would make sense to have a semi-open list of suggested values for this new "typ-ie" (`@dirtype` seems to dirty, doesn't it?) thing, just to make sure that things like `lv`are used consistently. Compiling suggestions for that list is probably a different issue, though…

**Comment by pe-ro** [comment]
See my comments on #712.

**Comment by rettinghaus** [comment]
Continuation of laissez vibrer: How should this be encoded?
![image](https://user-images.githubusercontent.com/7693447/175255154-fcc633b7-95ca-462c-a764-dcda1d516ad1.png)

Has anyone an idea for this? In a previous measure these are connected to noteheads in a chord, so this could probably be encoded with `@prev` and `@next`. But I have no clue how this could be encoded if you just want to have this very snippet encoded. How should those bows get their vertical /sounding position?

**Comment by ahankinson** [comment]
I'm assuming that if there's a "Lassez vibrer" there should be a note or an instrument that you're "Laissez vibrer"ing. So you would probably need, at a minimum, an initial scoredef with some instrument and some pitch values. This doesn't exist in isolation.

**Comment by pe-ro** [comment]
One way to encode these curves would be to use `@x`, `@y`, `@x2`, and `@y2` coordinates to specify the start (x,y) and end (x2, y2) points of the curve.  This way doesn't require encoding the notes, but it does mean that you must know the starting and ending points of the curves in the output coordinate system.

Another method would be to encode the notes with which the curves are associated, but hide them with `@visible="false"`.

**Comment by kepper** [comment]
according to @lpugin, this is related to types of `<dir>`. I think this means he volunteers himself…

**Comment by lpugin** [comment]
Allowing textual content within `<lv>` seems to be a reasonable first step (moving to MEI 5.0). More complex cases such the last one brought by @rettinghaus would still need to be thought further.

**Comment by ahankinson** [comment]
Textual content directly, or `<dir>`?

The advantage of allowing something like `<dir>` is that we don't need to add any extra attributes to `<lv>` directly to control rendering or fonts, etc.

**Comment by lpugin** [comment]
> Textual content directly, or `<dir>`?

Textual content directly. Starting to have control events within control events will cause all sorts of problems. Most of the text rendering control happens in `<rend>`, which would be allowed in `<lv>`. So if you want to change the font, you would to
```xml
<lv>
  <rend fontweight="bold">laissez vibrer</rend>
</lv>
```
I prefer it to:
```xml
<lv>
  <dir>
    <rend fontweight="bold">laissez vibrer</rend>
  </dir>
</lv>
```

**Comment by pe-ro** [comment]
This is another example of creating elements for subclasses of `<dir>`.  It's not a problem to put all the attributes of `<dir>` on `<lv>`.  So, I'm going to reverse my position and agree that direct textual content is fine.
